src/verify_multiple_aggregate_signatures.cc

```diff
@@ -0,0 +1,207 @@
+                return scope.Escape(Napi::Boolean::New(env, result));
+            case blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN:
+                return scope.Escape(env.Undefined());
+            case blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR:
+                Napi::Error::New(env, error_msg).ThrowAsJavaScriptException();
+                return scope.Escape(env.Undefined());
+            default:
+            case blst_ts::BLST_TS_ERROR::INVALID:
+                return scope.Escape(Napi::Boolean::New(env, false));
+        }
+    } catch (...) {
+        return Napi::Boolean::New(env, false);
+    }
+}
+
+class VerifyMultipleAggregateSignaturesWorker : public Napi::AsyncWorker {
+   public:
+    VerifyMultipleAggregateSignaturesWorker(const Napi::CallbackInfo &info)
+        : Napi::
+              AsyncWorker{info.Env(), "VerifyMultipleAggregateSignaturesWorker"},
+          deferred{Env()},
+          has_error{false},
+          _module{Env().GetInstanceData<BlstTsAddon>()},
+          _ctx{new blst::Pairing(true, _module->dst)},
+          _sets{},
+          _sets_ref{Napi::Persistent(info[0])},
+          _is_invalid{false},
+          _result{false} {
+        try {
+            blst_ts::BLST_TS_ERROR error =
+                prepare_verify_multiple_aggregate_signatures(_sets, info);
+            switch (error) {
+                case blst_ts::BLST_TS_ERROR::SUCCESS:
+                    return;
+                case blst_ts::BLST_TS_ERROR::JS_ERROR_THROWN:
+                    has_error = true;
+                    return;
+                default:
+                case blst_ts::BLST_TS_ERROR::INVALID:
+                    _is_invalid = true;
+            }
+        } catch (...) {
+            _is_invalid = true;
+        }
+    }
+
+    /**
+     * GetPromise associated with deferred for return to JS
+     */
+    Napi::Promise GetPromise() { return deferred.Promise(); }
+
+   protected:
+    void Execute() {
+        if (_is_invalid) {
+            return;
+        }
+        std::string error_msg{};
+        blst_ts::BLST_TS_ERROR error = verify_multiple_aggregate_signatures(
+            _result, error_msg, _module, _ctx, _sets);
+        if (error == blst_ts::BLST_TS_ERROR::HAS_NATIVE_ERROR) {
+            SetError(error_msg);
+        }
+    }
+    void OnOK() { deferred.Resolve(Napi::Boolean::New(Env(), _result)); }
+    void OnError(const Napi::Error &err) { deferred.Reject(err.Value()); }
+
+   public:
+    Napi::Promise::Deferred deferred;
+    bool has_error;
+
+   private:
+    BlstTsAddon *_module;
+    std::unique_ptr<blst::Pairing> _ctx;
+    std::vector<SignatureSet> _sets;
+    Napi::Reference<Napi::Value> _sets_ref;
+    bool _is_invalid;
+    bool _result;
+};
+
+Napi::Value AsyncVerifyMultipleAggregateSignatures(
+    const Napi::CallbackInfo &info) {
+    VerifyMultipleAggregateSignaturesWorker *worker =
+        new VerifyMultipleAggregateSignaturesWorker(info);
+    if (worker->has_error) {
```

Comment:
Would it make sense to validate the `info` first and then allocate or fail? This would get rid of a few members in this class.

Reply:
Yep.  Now that there is helper function for that it makes sense!  Good catch!

Reply:
Looked at this again and that is tricky.  Some of the conditions return `false` and not throw errors.  In order to return a result it needs to be wrapped in a promise. I attempted to make this work but it proved to be challenging and isnt worth it to save a couple member variables that just get immediately GC'd.

Reply:
Ok.... Second pass and got it figured out. Updated.